problems using linear-base in a new project

I've followed the steps from the readme in a new project, and I get the following error:

```
02:12 PM noon ∈ linear-types stack build
Stack has not been tested with GHC versions above 8.10, and using 8.11, this may fail
Stack has not been tested with Cabal versions above 3.2, but version 3.3.0.0 was found, this may fail
WARNING: Ignoring vector's bounds on base (>=4.5 && <4.15); using base-4.15.0.0.
Reason: allow-newer enabled.
WARNING: Ignoring primitive's bounds on base (>=4.5 && <4.15); using base-4.15.0.0.
Reason: allow-newer enabled.
primitive      > configure
primitive      > Configuring primitive-0.7.0.0...
primitive      > build
primitive      > Preprocessing library for primitive-0.7.0.0..
primitive      > Building library for primitive-0.7.0.0..
primitive      > [ 1 of 13] Compiling Control.Monad.Primitive
primitive      > 
primitive      > /run/user/1000/stack-fc9f49f9c873bb1e/primitive-0.7.0.0/Control/Monad/Primitive.hs:27:21: error:
primitive      >     Module ‘GHC.Base’ does not export ‘unsafeCoerce#’
primitive      >    |
primitive      > 27 | import GHC.Base   ( unsafeCoerce#, realWorld# )
primitive      >    |                     ^^^^^^^^^^^^^
            
--  While building package primitive-0.7.0.0 using:
      /home/noon/.stack/setup-exe-cache/x86_64-linux-nix/Cabal-simple_mPHDZzAJ_3.3.0.0_ghc-8.11.20200630 --builddir=.stack-work/dist/x86_64-linux-nix/Cabal-3.3.0.0 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
Progress 1/4
```

**To Reproduce**

1. Start a new project with `stack new linear-types --bare simple-hpack`
2. Follow the README - https://github.com/tweag/linear-base#getting-started
3. Build with `stack build`


**Expected behavior**

The build to work!

**Environment**
 - OS name + version: Linux 4.15.0-109-generic #110-Ubuntu SMP Tue Jun 23 02:39:32 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
 - Version of the code: Tried with the revision in the readme, as well as `705522f0bad3f1083bb3447b9f476f6f84d21410`

--- a/README.md
+++ b/README.md
@@ -40,8 +40,8 @@ version of GHC with support for linear types.
    commit in the `extra-deps`._
 
 ```yaml
-resolver: lts-14.6
-compiler: ghc-8.9
+resolver: lts-16.2
+compiler: ghc-8.11
 allow-newer: true
 system-ghc: true
 
@@ -51,9 +51,12 @@ nix:
   path: ["nixpkgs=./nixpkgs.nix"]
 
 extra-deps:
-  - primitive-0.7.0.0
+  - git: https://github.com/facundominguez/quickcheck.git
+    commit: a498e7b41131cf7955b9e154ab26d37d1be10304
+  - git: https://github.com/facundominguez/lifted-async.git
+    commit: 898eb485b21cc321058345998eedd8c409c326fd
   - git: https://github.com/tweag/linear-base.git
-    commit: 5dcb68d52229753f381110e8b0bb681245080235
+    commit: 705522f0bad3f1083bb3447b9f476f6f84d21410
 ```
 
 4. Add `linear-base` to your cabal file's `build-depends:` for the appropriate

